PROPOSALS.md

```diff
@@ -54,6 +62,15 @@ The Core Libraries Committee's focus is around management,
 not implementation: you should do the work for your own proposals
 (or others' if you like them!) to show us that you think it's worth doing.
 
+To be more precise, as the proposal author, you're expected to do the following
+(or find volunteers who'll do this for you):
+
+1. Refine your proposal as per the CLC requests.
+2. Make impact assessment in case of breaking changes.
+3. Write a migration guide.
+4. Submit forward compatibility patches to all impacted libraries.
```

Comment:
If 3. and 4. are indeed a hard requirement, then you'll need a label for proposals which ware blocked on that.

And maybe also a label / intermediate step of soft approval. I wouldn't go through the trouble of writing migration guide and especially compatibility patches if there it's unclear if the proposal will be accepted at all.

Reply:
I don't think these are hard requirements at all.

CLC has not established a baseline for these things yet, afaiu, so I'd say this document should be as general as possible without making assumptions about how CLC members will vote or what they will demand in order to vote +1.

Impact assessment is general enough that everyone can get behind, I think.

Reply:
@phadej That makes total sense 😌 I will change the wording to mention that you'll need to write a migration guide and submit patches only after your proposal is approved.

<hr>

@hasufell I agree with you and I don't think that having or not having impact assessment / migration guides / submitted patches should affect vote. However, after the impact assessment it may become clear that the proposed change breaks lots of stuff and this **actually may** affect the vote. So I would say that performing an impact assessment **is** hard requirement for breaking changes.

Reply:
current proposals.md says differently though:

> Once everything that breaks receives a patch, let us know in a comment where they've been submitted. Then, the committee will be able to vote!

Implying patches need to be written prior the vote.

I guess, I'm not going to propose any breaking changes.

Reply:
Preparing _some sort of_ patches is a necessary part of impact assessment: we cannot assess the exact amount of breakage unless there is a patch which allows compilation to pass. However, such patches are allowed to be fairly crude and not suitable for merging (e. g., not backwards-compatible).

Migration guide indeed is required only after approval. 

There is a certain ambiguity around "submit" verb. Note the current wording is "let us know in a comment where they've been submitted", meaning that it's fine to submit patches as a gist somewhere, not necessarily raise PRs. This is intentional. Partly because sometimes a package has no public repository, or it's not on GitHub, or it's not even Git-based; partly because package maintainers cannot (or are usually unwilling to) assess a PR, providing compatibility with a future, unreleased version of `base`. It's important to retain this subtlety of meaning in the updated process documentation.

Reply:
Yeah, I wouldn't want to have high barrier to writing proposals and trying to make improvements.

Certain changes with big negative impact would call for greater effort on the proposal author to ease that impact, but I wouldn't want someone to feel like they need to fork `base` and see how Stackage responds to a change *before* asking if it is a reasonable idea in the first place.